docs/admin/executors/deploy_executors_kubernetes.mdx

```diff
@@ -1,67 +1,48 @@
-for details.
+See the [example Role YAML](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/cmd/executor/kubernetes/batches/executor-batches.Role.yml) for more details on the exact RBAC requirements for native Kubernetes Executors.
 
-### Docker Image
+## Deployment
+
+Native Kubernetes Executors can be deployed via either the `sourcegraph-executor-k8s` [Helm chart](https://github.com/sourcegraph/deploy-sourcegraph-helm/tree/main/charts/sourcegraph-executor/k8s) or by using the `executors/k8s` [Kustomize component](https://github.com/sourcegraph/deploy-sourcegraph-k8s/tree/main/components/executors/k8s).
+
+### Steps
+
+1. If you are deploying Executors for processing Batch Changes, set the `native-ssbc-execution` [feature flag](/admin/executors/native_execution#enable)
+2. Configure the following environment variables on the Executor Deployment:
+    1. `EXECUTOR_FRONTEND_URL` should match the URL of your Sourcegraph instance
+    2. `EXECUTOR_FRONTEND_PASSWORD` should match the `executors.accessToken` key in the Sourcegraph instance's site configuration
+    3. Either `EXECUTOR_QUEUE_NAMES` or `EXECUTOR_QUEUE_NAME` should be set depending on whether the Executor will process batch change or precise auto indexing jobs
+    4. `EXECUTOR_KUBERNETES_NAMESPACE` should be set to the Kubernetes namespace where you intend to run the worker pods. This should generally match the namesapce where you deploy the Executor resources in the next step.
+
+    <Callout type="note">Additional environment variables may need to be configured for your Executors deployment. See [here](/admin/executors/executors_config) for a complete list.</Callout>
+
+3. Deploy the native Kubernetes Executor resources to your Kubernetes cluster via either the `sourcegraph-executor-k8s` [Helm chart](https://github.com/sourcegraph/deploy-sourcegraph-helm/tree/main/charts/sourcegraph-executor/k8s) or the `executors/k8s` [Kustomize component](https://github.com/sourcegraph/deploy-sourcegraph-k8s/tree/main/components/executors/k8s).
+    1. For more details on how to work with the Sourcegraph Helm chart, see the [Helm chart docs](/admin/deploy/kubernetes#configuration) (_Note: The same Helm values `override.yaml` file can be used for both the main `sourcegraph` Helm chart as well as the `sourcegraph-executor-k8s` chart_)
+    2. For more details on how to configure the `executors/k8s` component for your Kustomize deployment, see the [Kustomize configuration docs](/admin/deploy/kubernetes/configure#overview)
+4. Once all the native Kubernetes Executor resources have been deployed to your cluster, confirm that the Executors are online by checking the _Executors_ page under **Site admin > Executors > Instances**
 
-The Executor Docker image is available on Docker Hub
-at [`sourcegraph/executor-kubernetes`](https://hub.docker.com/r/sourcegraph/executor-kubernetes/tags).
-
-### Environment Variables
```

Comment:
Maybe add a link here to the new `## Kubernetes Specific Configuration` section in `docs/admin/executors/executors_config.mdx`. Having them all in one place is great, but admins will probably come here looking for them, so a link would be helpful

Reply:
There's a `Callout` component on line 31 that links to the configuration doc. I placed it right under the second step where we mention the "mandatory" configuration that the reader will have to fill out.

Reply:
<img width="864" alt="Screenshot 2024-06-13 at 23 24 38" src="https://github.com/sourcegraph/docs/assets/31862633/c3d7bbc7-1a25-45d4-be6d-eef8aff478f9">